lxd/instance/drivers/driver_qemu.go

```diff
@@ -1565,6 +1565,13 @@ func (d *qemu) Start(stateful bool) error {
 		return err
 	}
 
+	go func() {
+		err := d.AdvertiseVMSockAddress()
```

Comment:
Rather than waiting for instance to start up, why don't we include the current LXD connection info in the config files used by lxd-agent when it starts up?

Reply:
> Rather than waiting for instance to start up, why don't we include the current LXD connection info in the config files used by lxd-agent when it starts up? 

One could have put that in the `instance-data` file, but we want to remove that one.

> Would save having these go routines hanging around for a (potentially) long time.

I don't consider a maximum of 60 seconds a long time ;)

Reply:
If you have hundreds of VMs, all starting up, and these go routines hanging around for 60s trying repeatedly to connect to LXD agent, then that is potentially lot of busy work. OK, if not putting it in the instance-data file, how about triggering this process of checking the lxd-agent when it tells us its running (via the existing `STARTED` ring-buf state)?

Reply:
> If you have hundreds of VMs, all starting up, and these go routines hanging around for 60s trying repeatedly to connect to LXD agent, then that is potentially lot of busy work.

`AdvertiseVsockAddress()` doesn't repeatedly try to connect. It waits until the agent is ready before attempting to connect.

> OK, if not putting it in the instance-data file, how about triggering this process of checking the lxd-agent when it tells us its running (via the existing STARTED ring-buf state)?

How is this different than the current state?

Reply:
That sounds like a good idea. I'll give it a try.